### PR TITLE
tests(gh): increase cli test timeout to reduce flakes

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
The CLI run-test keeps timing out in GH actions and failing, so why not up it? Also reduces it down to *just* PRs (that's what we do for travis, so why do push *and* PR for GH?) though I don't feel as strongly about that one.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 name: 💡🏠
 
-on: [push, pull_request]
+on: pull_request
```
Comment (patrickhulce) [inline .github/workflows/ci.yml:7]:
```suggestion
on: [push, pull_request]
```

**Comment by patrickhulce** [comment]
aight I'm fine to defer that 👍

**Comment by patrickhulce** [comment]
Ah so now the truth comes out!

https://github.com/GoogleChrome/lighthouse/runs/615776700?check_suite_focus=true#step:16:263

image

maybe we need to just retry it...

**Comment by connorjclark** [comment]
that's super odd.

https://github.com/bluzi/jest-retries ?

**Comment by patrickhulce** [comment]
jest actually has retry support built-in if we update to their circus runner, it's pretty nice lighthouse-ci is on it :)

**Comment by patrickhulce** [comment]
@connorjclark how do you feel about landing the timeout change at least to track the frequency of unable to connect? from other PRs this doesn't seem as big of a deal as before...

**Pull request merged**